docs/modules/user-guide/partials/proc_developing-with-devfiles.adoc

```diff
@@ -0,0 +1,35 @@
+[id="proc_developing-with-devfiles_{context}"]
+= Developing with devfiles
+[role="_abstract"]
+
+Use the devfile specification to develop a node.js “Hello World” application. Developing this application introduces you to how a devfile automates and simplifies your development process.
+
+.Prerequisites
+
+* To develop an application by using the devfile specification, you need to create a Kubernetes cluster on your local machine. To create such a cluster, install link:https://minikube.sigs.k8s.io/docs/start/[minikube start].
+* To access your namespaces, download link:https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/[kubectl].
+* When you use minikube for the first time, you must enable Ingress to access your devfile projects. Enable Ingress by running `minikube addons enable ingress`.
+* To execute the devfile specification, install the link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.6/html/cli_tools/developer-cli-odo#installing-odo[odo cli tool].
+
+.Procedure
+
+. Run `minikube start` to build your Kubernetes cluster.
+. Run `kubectl get namespaces` to confirm you are in the default namespace.
+. Run `odo catalog list components` to view the available devfiles.
+. Run `odo create nodejs <name of your project> --starter` to create a devfile project.
+* Adding `--starter` tells odo to include the starter project currently inside the nodejs devfile specification, which makes it easier for you to develop an application.
+. Run `odo push`. Doing so instructs odo to create the Kubernetes objects specified in the devfile.
+. Run `kubectl get pod` to access your container.
+. Run `minkube IP` to find your cluster IP address.
+. Run `odo url create <name you give the url> --ingress --host <IP address>.nip.io`. Doing so creates an ingress inside your cluster that you can use to access your application using the cluster IP address.
+* Example: `odo url create myfirstproject --ingress --host 192.168.64.2.nip.io`.
+. Run `odo push` to build the URL.
```

Comment:
any specific reason we want to split into two `push` cmds?

Reply:
@yangcao77 I thought we needed to run `odo push` again to build the URL that we created with odo? Is the last step, line 26, `run odo push to build the URL` unnecessary, and therefore, can be deleted?